spicy/runtime/src/parser.cc

```diff
@@ -29,7 +29,7 @@ static bool _haveEod(const hilti::rt::ValueReference<hilti::rt::Stream>& data, c
     if ( cur.isOpenEnded() )
         return false;
 
-    return cur.unsafeEnd().offset() <= data->unsafeEnd().offset();
+    return cur.end().offset() <= data->end().offset();
```

Comment:
Yeah, this looks right, given the implementation of  `end()`. I'm wondering if we have more cases like this elsewhere. The unsafe itereators are supposed to be more efficient than the safe versions, so the code usually prefers using/creating them.

Reply:
When making this patch I looked at other users of `unsafeEnd` and this one seemed the most problematic since it is repeatedly called from generated code. There are other questionable uses of `unsafeEnd`, but they didn't seem to be so easy to trigger and also didn't show up in profiling output for test runs. I still filed #1347 to potentially clean up this API since it seems easy to introduce performance problems.